cannon/mipsevm/memory/memory.go

```diff
@@ -230,8 +233,15 @@ func (m *Memory) AllocPage(pageIndex Word) *CachedPage {
 	p := new(CachedPage)
 	for _, region := range m.MappedRegions {
 		if region.PageIndexInRegion(pageIndex) {
-			indexAdjusted := pageIndex - region.startAddr>>PageAddrSize
-			p.Data = region.Data[indexAdjusted*PageSize : (indexAdjusted+1)*PageSize : (indexAdjusted+1)*PageSize]
+			indexAdjusted := pageIndex - (region.startAddr >> PageAddrSize)
+			start := indexAdjusted * PageSize
+			end := (indexAdjusted + 1) * PageSize
+			// Bounds check to avoid panic
+			if end <= Word(len(region.Data)) && start < end { //start >= 0 not needed as start is unint and always >= 0
+				p.Data = region.Data[start:end:end]
+			} else {
+				p.Data = make(Page, PageSize)
```

Comment:
Similar to my above comment, we should not be expecting an out of bounds access due to the check on L235.